dart/lib/src/scope.dart

```diff
@@ -142,6 +142,59 @@ class Scope {
   /// Removes an extra from the Scope
   void removeExtra(String key) => _extra.remove(key);
 
+  SentryEvent applyToEvent(SentryEvent event, dynamic hint) {
+    if (event.transaction == null) {
+      event = event.copyWith(transaction: transaction);
+    }
+
+    // Merge the user context.
+    if (event.userContext == null) {
+      event = event.copyWith(userContext: user);
+    }
+
+    // Merge the scope fingerprint.
+    if (event.fingerprint == null) {
+      event = event.copyWith(fingerprint: fingerprint);
+    }
+
+    // Merge the scope breadcrumbs.
+    if (event.breadcrumbs == null) {
+      event = event.copyWith(breadcrumbs: breadcrumbs);
+    }
+
+    // Merge the scope tags.
+    event = event.copyWith(
+        tags: tags.map((key, value) => MapEntry(key, value))
+          ..addAll(event.tags ?? {}));
+
+    // Merge the scope extra.
+    event = event.copyWith(
+        extra: extra.map((key, value) => MapEntry(key, value))
+          ..addAll(event.extra ?? {}));
```

Comment:
if both tags have the same key, which one wins with this syntax? worth a comment, probably the `event` one should be kept.

Reply:
✅